gst/nnstreamer/tensor_query/tensor_query_client.c

```diff
@@ -342,25 +342,49 @@ gst_tensor_query_client_get_property (GObject * object, guint prop_id,
 /**
  * @brief Update src pad caps from tensors config.
  */
-static void
+static gboolean
 gst_tensor_query_client_update_caps (GstTensorQueryClient * self,
     const gchar * caps_str)
 {
   GstCaps *curr_caps, *out_caps;
-
+  gboolean ret = FALSE;
   out_caps = gst_caps_from_string (caps_str);
 
   /* Update src pad caps if it is different. */
   curr_caps = gst_pad_get_current_caps (self->srcpad);
   if (curr_caps == NULL || !gst_caps_is_equal (curr_caps, out_caps)) {
-    silent_debug_caps (self, out_caps, "set out-caps");
-    gst_pad_set_caps (self->srcpad, out_caps);
+    GstCaps *sink_caps;
+    GstStructure *structure;
+    gint rate_n = 0, rate_d = 1;
+
+    /** Get framerate from source pad caps */
+    sink_caps = gst_pad_get_current_caps (self->sinkpad);
+    if (sink_caps) {
+      structure = gst_caps_get_structure (sink_caps, 0);
+      if (gst_structure_has_field (structure, "framerate")) {
+        gst_structure_get_fraction (structure, "framerate", &rate_n, &rate_d);
+      }
+      gst_caps_unref (sink_caps);
+    }
+
+    out_caps = gst_caps_make_writable (out_caps);
+    gst_caps_set_simple (out_caps, "framerate", GST_TYPE_FRACTION, rate_n,
```

Comment:
Q: Is it necessary to set framerate on srcpad?
src pad caps template is any. plz consider the case if src is audio or other mimetype that does not need framerate.